File: apollo-api/src/main/java/com/apollographql/apollo/api/internal/QueryDocumentMinifier.java
package com.apollographql.apollo.api.internal;

import org.jetbrains.annotations.NotNull;

public final class QueryDocumentMinifier {

  private QueryDocumentMinifier() {
  }

  public static String minify(@NotNull String queryDocument) {
    return queryDocument.replaceAll("\\s *", " ");
  }
}

File: libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/QueryDocumentMinifier.kt
package com.apollographql.apollo3.api

import com.apollographql.apollo3.annotations.ApolloDeprecatedSince
import kotlin.jvm.JvmStatic

@Deprecated("Use the version in apollo-ast instead or copy paste this implementation", ReplaceWith("com.apollographql.apollo3.ast.QueryDocumentMinifier"))
@ApolloDeprecatedSince(ApolloDeprecatedSince.Version.v3_3_1)
object QueryDocumentMinifier {
  @JvmStatic
  fun minify(queryDocument: String): String {
    return queryDocument.replace("\\s *".toRegex(), " ")
  }
}
